### PR TITLE
Added POST following like friendrequest

### PR DESCRIPTION
**Pull request by mraypold**
Makes following API the same as friending. Also removes ModifyRelationsMixin that wasn't being used.

```
"author":{
  "id":"bob",
  "host":"http://example.org",
  "displayname":"bsmith",
},
"following":{
  "id":"beck",
  "host": "http://example.com",
  "displayname":"bcent",
}
```

### CODE REVIEW

File: api/views/author.py [hunk 1]
```diff
@@ -113,43 +53,32 @@ def post(self, request, *args, **kwargs):
...
-            # Who we are following
-            follow = get_object_or_404(self.queryset, id=pk)
-            author.follow(follow)
-        else:
-            author = get_object_or_404(self.queryset, id=pk)
+    SAFE_METHODS = ['GET',]
 
+    # GET author/:author_pk/follow
+    # Returns who author_pk following
+    def list(self, request, author_pk=None):
```
Comment (xkb) [inline api/views/author.py]: is this exposed to everyone authenticated?

Comment (mraypold) [inline api/views/author.py]: This can be accessed by anyone, even if not authenticated.

Comment (xkb) [inline api/views/author.py]: Lets not have this call at all. `GET /author/:id` returns the appropriate lists based on permission.

Comment (mraypold) [inline api/views/author.py]: It's now been removed.

**Pull request merged**